### PR TITLE
[15.0][FIX]account_move_line_sale_info: picke info directly form sale line ids field

### PR DESCRIPTION
**Pull request by AaronHForgeFlow**
forward port of https://github.com/OCA/account-financial-tools/pull/1570

cc @ForgeFlow

### CODE REVIEW

File: account_move_line_sale_info/models/account_move.py [hunk 1]
```diff
@@ -10,17 +10,10 @@ class AccountMove(models.Model):
...
         res = super()._stock_account_prepare_anglo_saxon_out_lines_vals()
         for i, vals in enumerate(res):
-            if (
-                not vals.get("move_id", False)
-                or not vals.get("product_id", False)
-                or not vals.get("quantity", False)
-            ):
+            if not vals.get("move_id", False):
                 continue
             am = self.env["account.move"].browse(vals["move_id"])
```
Comment (MiquelRForgeFlow) [inline account_move_line_sale_info/models/account_move.py]:
```suggestion
```

Comment (AaronHForgeFlow) [reply account_move_line_sale_info/models/account_move.py]: ok

File: account_move_line_sale_info/models/account_move.py [hunk 2]
```diff
@@ -10,17 +10,8 @@ class AccountMove(models.Model):
...
-                or not vals.get("product_id", False)
-                or not vals.get("quantity", False)
-            ):
-                continue
             am = self.env["account.move"].browse(vals["move_id"])
-            sale_line_id = am.invoice_line_ids.filtered(
-                lambda il: il.product_id.id == vals["product_id"]
-                and il.quantity == vals["quantity"]
-            ).mapped("sale_line_id")
+            sale_line_id = am.sale_line_ids
```
Comment (MiquelRForgeFlow) [inline account_move_line_sale_info/models/account_move.py]:
```suggestion
            sale_line_id = am.invoice_line_ids.filtered(
                  lambda il: il.product_id.id == res["product_id"]
                  and il.quantity == res["quantity"]
              ).mapped("sale_line_ids")
```

Comment (AaronHForgeFlow) [reply account_move_line_sale_info/models/account_move.py]: good point

**Review by MiquelRForgeFlow — APPROVED** [review]

**Comment by AaronHForgeFlow** [comment]
/ocabot merge patch

keeping the branches aligned

**Pull request merged**